c_src/main.c

```diff
@@ -86,23 +86,16 @@ erlfdb_future_cb(FDBFuture* fdb_future, void* data)
     cancelled = future->cancelled;
     enif_mutex_unlock(future->cancel_lock);
```

Comment:
Stupid question, do we released the lock too early? 
image two process is cancelling the future if it can happen?

Reply:
fdb allows double cancelling; it only forbids double destroy

Reply:
If FDB ensures the callback is executed exact once as the cancel could be triggered by FDB or by NIF.
and the deref of future NIF resource is only done (for NIF part) in the callback  https://github.com/emqx/couchdb-erlfdb/pull/22/files#diff-ceadfa165520612e5b6b1a255f7680bcaa9e2a67bf47677f6884fd1db3cb03f5L110 .
Then I think the future->cancel_lock is not needed at all?

Reply:
it is unrelated to this PR though.

Reply:
`future->cancelled` is independently set from two different nifs;  so, formally, we need synchronization